### DISCUSSION TITLE
Separate framework just for the core part

### DISCUSSION THREAD

**Discussion by nonameplum** [pointfreeco/swift-composable-architecture#1694] [Ideas]
Hey 👋,
Did you consider to take out the core part of the framework like Store, Reducer without UI part so it could be used as kinda state machine where some would like to use it to implement a logic only?
Probably it could be universal framework regardless the Apple platform.

**Comment by stephencelis** [comment]
Hi @nonameplum!

We'd definitely like to make TCA more and more portable. Currently its core is driven using Combine, though, which would require adopting something like OpenCombine or something else. It's something we'd definitely like to look into, though we have a few things that are higher-priority for us right now for [our 1.0 release](https://github.com/pointfreeco/swift-composable-architecture/discussions/1477).

This kind of portability is definitely something we're looking to support in 2.0, though!

**Comment by nonameplum** [comment]
Sounds great 💯
